python/ray/_private/node.py

```diff
@@ -206,8 +185,39 @@ def __init__(
                     f"{ray_params.dashboard_host}:{ray_params.dashboard_port}"
                 )
 
+        # It creates a session_dir.
         self._init_temp()
 
+        node_ip_address = ray_params.node_ip_address
+        if connect_only:
+            self.wait_for_node_address()
+        else:
+            if node_ip_address is None:
+                node_ip_address = ray._private.services.resolve_ip_for_localhost(
+                    ray_constants.NODE_DEFAULT_IP
+                )
+
+        node_ip_address = self._get_or_write_node_address(node_ip_address)
+
+        ray_params.update_if_absent(
+            node_ip_address=node_ip_address, raylet_ip_address=node_ip_address
+        )
+
+        self._node_ip_address = node_ip_address
+
+        if ray_params.raylet_ip_address:
+            raylet_ip_address = ray_params.raylet_ip_address
+        else:
+            raylet_ip_address = node_ip_address
+
+        if raylet_ip_address != node_ip_address and (not connect_only or head):
```

Comment:
With this PR, do we still supporting raylet_ip and node_ip being different?

Reply:
what's the case that this happens?